### ISSUE TITLE
Actuator cannot record metrics for downstream asynchronous tasks

### ISSUE THREAD

**Issue by thomasabraham** [zalando/zmon-actuator#20]
I have an application that uses RxJava/Jersey/Spring boot. 
I have the zmon-actuator metrics displayed on a grafana board. 
As I noticed my actual latency didn't match with what is shown in grafana board, I looked into it and found a problem with the use of `ZmonMetricsFilter extends OncePerRequestFilter`. 
The stopwatch is started, my application logic triggers some external calls, stop watch is stopped and the metrics recorded, now my application logic get callbacks from external services and it sends final response to the client. 
Basically I'm suing @Suspended annotation of jax-rs

**Comment by jbellmann** [comment]
@thomasabraham thanks for reporting this, can we have a look into your example/usecase? To get a better picture of that?

**Comment by thomasabraham** [comment]
As I mentioned I'm using RxJava/Jersey/Spring boot stack and use @suspended annotation to make things async.

Consider an API which simply take a record from a data source and return to client. 

In a traditional synchronous straightforward app, when it get a request, it is passed on to a controller function after creating a new thread. And this thread will be used to process the request and return the result. And the once per request filter will be executed before and after the request as expected. 

1. Got request
2. stop watch is started
3. Controller function is called
4. Request is processed
5. Controller function is returned
6. stop watch is stopped
7. Response is send

In my use case I need to handle very high number of connections in parallel and each requests will make calls to external services; hence they take 200-500 ms to complete processing. So keeping a thread open for every request I have to process is not very efficient. So I use the RxJava to make this asynchronous and this is how things will be.

1. Got request
2. stop watch started
3. controller function is called
4. A task is created with the http request/connection object and started
5. controller function is returned
6. stop watch is stopped
7. the task is executed is some thread in a non blocking way
8. when the response is ready, it is send back to the client. 

And to fix this and make it work we should have an option to use jersey filter instead of springs OncePerRequestFilter. As this filter will be executed only when the actual response is send back.

**Comment by suspended** [comment]
Did someone tag me?

**Comment by beverage** [comment]
@suspended Lol @ Java conventions collisions!  Nothing to see here for you.  :)

@thomasabraham If this is still something you want to do, it sounds more like an OpenTracing use case than a ZMON use case.  Please reach out to Team EagleEye if so.
